processor/spanmetricsprocessor/processor.go

```diff
@@ -92,14 +95,14 @@ type processorImp struct {
 	startTime time.Time
 
 	// Call & Error counts.
-	callSum map[resourceKey]map[metricKey]int64
+	callSum map[resourceKey]map[instrLibKey]map[metricKey]int64
 
 	// Latency histogram.
-	latencyCount         map[resourceKey]map[metricKey]uint64
-	latencySum           map[resourceKey]map[metricKey]float64
-	latencyBucketCounts  map[resourceKey]map[metricKey][]uint64
+	latencyCount         map[resourceKey]map[instrLibKey]map[metricKey]uint64
```

Comment:
I understand why you've done it this way, but it does make it more complex. It seems the more we add to this the more we are just replicating what the metrics SDK is designed to do, except in a makeshift way.

Probably not necessary for now, but I would like to investigate if it's possible to simply use an SDK instrument and aggregator to create these metrics rather than re-inventing the wheel.

Reply:
Do you have any links to the stuff you mentioned, I'm not very familiar with it. Is there an in-built aggregator in the SDK? :curious:

Reply:
We should consider removing the aggregation from the processor. Instead, it should only focus on extraction. See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/403#issuecomment-1017021263

Reply:
Yeah, the other idea is connect it up to other metrics processor so it can handle the funky stuff there. I don't have any links, it's just in my head. But there's definitely a better way, like you mentioned in the linked comment.